### PR TITLE
feat: snapshots support for VC

### PR DESCRIPTION
**Pull request by tomas-danek**

### CODE REVIEW

File: vc-worker.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def enqueue_get_info_request(request, machine, conn):
     ).save(conn=conn)
 
 
-def action_take_screenshot(request, machine, vc, action, conn):
+def action_take_screenshot(request, machine, vc, conn):
```
Comment (tomas-danek) [inline vc-worker.py]: just removed action param

File: vc-worker.py [hunk 2]
```diff
@@ -198,6 +198,58 @@ def action_take_screenshot(request, machine, vc, action, conn):
...
+        snap.status = 'success' if result is True else 'failed'
+        snap.save(conn=conn)
+        if result is True:
+            # attach snapshot from machine if creating was successful
+            machine = data.Machine.get_one_for_update({'_id': machine.id}, conn=conn)
+            machine.snapshots.append(snap.id)
+            machine.save(conn=conn)
+
+    else:
+        settings.raven.captureMessage('Error obtaining subject_id from Request')
```
Comment (petrSchreiber) [inline vc-worker.py]: What do you think, would it add some value to have the message a bit more specific like:
`'Error obtaining request subject_id for take snapshot'`

File: web/module/snapshots.py [hunk 3]
```diff
@@ -1,8 +1,92 @@
...
-    return {'check': 'machines_get_snapshots {}, rq: {}'.format(machine_id, request.headers)}
+# @snapshots.route('/machines/<machine_id>/snapshots', methods=['GET'])
+# async def machines_deploy(request, machine_id):
+#     return {'check': 'machines_get_snapshots {}, rq: {}'.format(machine_id, request.headers)}
+
+@snapshots.route('/machines/<machine_id>/snapshots', methods=['POST'])
+async def take_snapshot(request, machine_id):
+    with data.Connection.use() as conn:
+
+        snapshot_name = request.headers['json_params']['name']
```
Comment (ireznice) [inline web/module/snapshots.py]: search for name within all snapshots and check name uniqueness

File: vc-worker.py [hunk 4]
```diff
@@ -198,6 +198,58 @@ def action_take_screenshot(request, machine, vc, action, conn):
...
+    if request.subject_id:
+        snap_ro = data.Snapshot.get_one({'_id': request.subject_id}, conn=conn)
+        result = vc.remove_snapshot(uuid=machine.provider_id, snapshot_name=snap_ro.name)
+        snap = data.Snapshot.get_one_for_update({'_id': request.subject_id}, conn=conn)
+        snap.status = 'success' if result is True else 'failed'
+        snap.save(conn=conn)
+        if result is True:
+            # detach snapshot from machine if remove was successful
+            machine = data.Machine.get_one_for_update({'_id': machine.id}, conn=conn)
+            machine.snapshots.remove(snap.id)
```
[the remark this answers is not in the document]
Comment (ireznice) [reply vc-worker.py]: SOLVED

**Pull request merged**